### DISCUSSION TITLE
Attach middleware to route groups

### DISCUSSION THREAD

**Discussion by notramo** [martenframework/marten#195] [Ideas]
For example, all `admin:` routes could have a middleware that checks user role, and all `api:` routes could have one that checks that accepted content-type is JSON.

**Comment by ellmetha** [comment]
The idea of middlewares is that they are applied to all incoming requests _before_ routes are actually resolved and _after_ a response has been produced by handlers (if a handler is indeed matched by the current route). If some handlers only need certain behaviors or checks, I would suggest making use of a base handler class or a shared concern module.

**Reply by notramo** [reply]
The problem with this approach is that it's easy to forget to add a module to a given route. Also, if a handler is moved to another route group, it's easy to forget to change the imported modules.
The base idea for this proposal is to attach middleware to the handlers, based on the route path.

However, if you think that it's out of scope for this project, it's not a must-have, but a nice-to-have indeed, so I'm closing this.
